backend/windmill-worker/src/worker.rs

```diff
@@ -1829,20 +1829,29 @@ async fn handle_queued_job<R: rsmq_async::RsmqConnection + Send + Sync + Clone>(
         None
     };
 
+    let (raw_code, raw_lock, raw_flow) =
+        sqlx::query!(
+            "SELECT raw_code, raw_lock, raw_flow AS \"raw_flow: Json<Box<RawValue>>\"
+            FROM queue_view WHERE id = $1 AND workspace_id = $2 LIMIT 1",
+            &job.id, job.workspace_id
+        )
+        .fetch_one(db)
+        .await
+        .map(|record| (record.raw_code, record.raw_lock, record.raw_flow))
+        .unwrap_or_default();
+
     let cached_res_path = if job.cache_ttl.is_some() {
         let version_hash = if let Some(h) = job.script_hash {
             format!("script_{}", h.to_string())
-        } else if let Some(rc) = job.raw_code.as_ref() {
+        } else if let Some(rc) = raw_code.as_ref() {
             use std::hash::Hasher;
             let mut s = DefaultHasher::new();
             rc.hash(&mut s);
             format!("inline_{}", hex::encode(s.finish().to_be_bytes()))
-        } else if let Some(rc) = job.raw_flow.as_ref() {
+        } else if let Some(sqlx::types::Json(rc)) = raw_flow.as_ref() {
             use std::hash::Hasher;
             let mut s = DefaultHasher::new();
-            serde_json::to_string(&rc.0)
-                .unwrap_or_default()
-                .hash(&mut s);
+            rc.get().hash(&mut s);
```

Comment:
@rubenfiszel I did modified this as `raw_flow` is already a string, but, maybe `serde_json::to_string` was used to min the json before hashing ?